tools/clang/lib/CodeGen/CGCleanup.cpp

```diff
@@ -942,15 +942,18 @@ bool CodeGenFunction::isObviouslyBranchWithoutCleanups(JumpDest Dest) const {
 /// be known, in which case this will require a fixup.
 ///
 /// As a side-effect, this method clears the insertion point.
-void CodeGenFunction::EmitBranchThroughCleanup(JumpDest Dest) {
+void CodeGenFunction::EmitBranchThroughCleanup(JumpDest Dest, llvm::BranchInst *WaveBr) {
   assert(Dest.getScopeDepth().encloses(EHStack.stable_begin())
          && "stale jump destination");
 
   if (!HaveInsertPoint())
     return;
 
   // Create the branch.
-  llvm::BranchInst *BI = Builder.CreateBr(Dest.getBlock());
+  // HLSL Change Begin - use pre-generated branch if exists
+  // If not a wave-enabled stage, we can keep everything unconditional as before
```

Comment:
I'm not sure the last line of this comment makes sense here. Even in a non-wave stage, WaveBr should be non-null.